src/magic/model/event/MagicSpellCardEvent.java

```diff
@@ -38,6 +40,9 @@ public static MagicSpellCardEvent create(final MagicCardDefinition cdef, final S
         if (cdef.hasAbility(MagicAbility.HauntSpell)) {
             return Haunt(rule);
         }
+        if (cdef.hasAbility(MagicAbility.Ascend)) {
```

Comment:
For spells, it will be simpler to do Ascent as part of the effect text, i.e. as an enum in MagicRuleEventAction rather than as a MagicSpellCardEvent. i.e. Ascent is part of the spell's effect like "Gain 5 life" kind of effect

Reply:
A similar effect would be MagicRulesEventAction.BecomesMonarch

Reply:
Fixed. I misunderstood the word "spell ability" in the comprehensive rules.

Reply:
Thanks for the change, it looks much better and simpler now. We try to put spell abilities in MagicRuleEventAction if possible, then it can also be reused as an effect of activated abilities. We only put it here in MagicSpellCardEvent as a fallback or it modifies something about the casting/resolution of the spell that cannot be represented in MagicRuleEventAction.

Reply:
I see. Thanks for the explanation!